packages/repository/test/unit/repository-mixin/repository-mixin.test.ts

```diff
@@ -42,6 +42,39 @@ describe('RepositoryMixin', () => {
     expectNoteRepoToBeBound(myApp);
   });
 
+  it('mixed class has .dataSource()', () => {
+    const myApp = new AppWithRepoMixin();
+    expect(typeof myApp.dataSource).to.be.eql('function');
+  });
+
+  it('binds dataSource from app.dataSource()', () => {
+    const myApp = new AppWithRepoMixin();
+
+    const withoutDataSource = myApp.find('datasource.*');
+    expect(withoutDataSource).to.be.empty();
+    // binding with name from juggler datasource
+    const fooDataSource: juggler.DataSource = new DataSourceConstructor({
+      name: 'foo',
+      connector: 'memory',
+    });
+    myApp.dataSource(fooDataSource);
+    expect(myApp.find('datasource.*').map(d => d.key)).to.containEql(
+      'datasource.foo',
+    );
+    expect(myApp.getSync(`datasource.foo`)).to.be.eql(fooDataSource);
+
+    // binding with explicit name
```

Comment:
One logical assertion per test please.

The code below should be extracted into a new test.

Reply:
Sure, I already refactored the block.